test/integration/master_replica_test.lua

```diff
@@ -0,0 +1,121 @@
+                    roles = { 'vshard-router' },
+                    servers = {
+                        { instance_uuid = helpers.uuid('a', 1), alias = 'router' },
+                    },
+                },
+                {
+                    uuid = helpers.uuid('b'),
+                    alias = 's-1',
+                    roles = { 'vshard-storage', 'customers-storage' },
+                    servers = {
+                        { instance_uuid = helpers.uuid('b', 1), alias = 's1-master' },
+                        { instance_uuid = helpers.uuid('b', 2), alias = 's1-replica' },
+                    }
+                }
+            },
+        })
+        cg.cluster:start()
+    end
+end)
+
+g.after_all(function(cg)
+    if is_cartridge_helpers then
+        cg.cluster:stop()
+        fio.rmtree(cg.cluster.datadir)
+    end
+end)
+
+g.before_each(function()
+    t.skip_if(not is_cartridge_helpers, "cartridge is not installed")
+end)
+
+g.after_each(function(cg)
+    cg.cluster:server('s1-master').net_box:eval([[
+        box.space.customers:truncate()
+    ]])
+end)
+
+local tuples_cnt = 3
+local function insert_tuples(cg)
+    cg.cluster:server('s1-master').net_box:eval([[
+        box.space.customers:insert({1})
+        box.space.customers:insert({2})
+        box.space.customers:insert({3})
+    ]])
+end
+
+local expirationd_eval = string.format([[
+    local deleted = 0
+    local task = require('expirationd').start("clean_all", 'customers',
+                                              function()
+                                                  deleted = deleted + 1
+                                                  return true
+                                              end,
+                                              {full_scan_delay = 0})
+    local retry = 100
+    for _ = 1, 100 do
+        if deleted == %d then
+            break
+        end
+        require("fiber").yield()
+    end
+    task:kill()
+    return #box.space.customers:select({}, {limit = 10})
+]], tuples_cnt)
+
+function g.test_master_is_enabled(cg)
+    insert_tuples(cg)
+    local ret = cg.cluster:server('s1-master').net_box:eval([[
+        return #box.space.customers:select({}, {limit = 10})
+    ]])
+    t.assert_equals(ret, tuples_cnt)
+
+    ret = cg.cluster:server('s1-master').net_box:eval(expirationd_eval)
+    t.assert_equals(ret, 0)
+end
+
+function g.test_replica_is_disabled(cg)
+    insert_tuples(cg)
+    local ret = cg.cluster:server('s1-master').net_box:eval([[
+        return #box.space.customers:select({}, {limit = 10})
+    ]])
+    t.assert_equals(ret, tuples_cnt)
+
+    -- wait tuples on the replica
+    local success_wait = false
+    for _ = 1, 30 do
+        local ret = cg.cluster:server('s1-replica').net_box:eval([[
+            return #box.space.customers:select({}, {limit = 10})
+        ]])
+        if ret == tuples_cnt then
+            success_wait = true
+            break
+        end
+        fiber.sleep(1)
```

Comment:
Tarantool replication is not so slow, I think it's better to set a smaller refresh timeout.

Reply:
It was rewritten to `helpers.retrying`.